rm_hw:
  bus:
    - can0
    - can1
  loop_frequency: 1000
  cycle_time_error_threshold: 0.001
  # Configurations of the actuators
  actuators:
    right_front_wheel_motor:
      bus: can0
      id: 0x201
      type: rm_3508
      lp_cutoff_frequency: 150
  #     need calibration: true  #set to true when this actuator need to be calibrated and it defaults to false when it is not been set.
  imus:
    gimbal_imu:
      frame_id: gimbal_imu
      bus: can1
      id: 0x300
      orientation_covariance_diagonal: [ 0.0012, 0.0012, 0.0012 ]
      angular_velocity_covariance: [ 0.0004, 0.0004, 0.0004 ]
      linear_acceleration_covariance: [ 0.01, 0.01, 0.01 ]
      angular_vel_coeff: 0.0010652644
      accel_coeff: 0.0017944335
      temp_coeff: 0.125
      temp_offset: 23.0
      filter: complementary
      do_bias_estimation: false
      do_adaptive_gain: true
      gain_acc: 0.0003
  gpios:
    front_lamp:
      pin: 245
      direction: out
    back_lamp:
      pin: 246
      direction: out
